src/hdf5/wsfm_2ioda.py

```diff
@@ -0,0 +1,310 @@
+
+    obs_key = ('brightnessTemperature', "ObsValue")
+    obs_data = set_missing_value(f, nchans, chk_geolocation, quality_word, obs_key, obs_data, skip=skip)
+
+    return obs_data
+
+
+def set_missing_value(f, nchans, chk_geolocation, quality_word, obs_key, obs_data, skip=1):
+
+    # use quality word to determine where to set for missing values
+    bands = f['BandFrequency']
+    fillvalue = f['TbHighRes'].fillvalue
+    for jchan, freq in enumerate(f['ChanFrequency']):
+        band_idx = np.where(bands == freq)[0][0]
+        i_pass = quality_word[:, band_idx]
+        chk_ob = (i_pass > 0) | (obs_data[obs_key][:, jchan] == fillvalue) | (chk_geolocation)
+        obs_data[obs_key][:, jchan][chk_ob] = float_missing_value
+
+    # this is a data reduction step that removes values that are missing
+    tb_key = 'brightnessTemperature'
+    good = (obs_data[(tb_key, obsValName)][:, 0] != float_missing_value) & \
+        (obs_data[(tb_key, obsValName)][:, 4] != float_missing_value) & \
+        (obs_data[(tb_key, obsValName)][:, 8] != float_missing_value) & \
+        (obs_data[(tb_key, obsValName)][:, 9] != float_missing_value) & \
+        (obs_data[(tb_key, obsValName)][:, 13] != float_missing_value)
+    for k in obs_data:
+        if metaDataName in k[1] and 'sensorChannelNumber' not in k[0]:
+            obs_data[k] = obs_data[k][good][::skip]
+        elif tb_key in k[0]:
+            obs_data[k] = obs_data[k][good, :][::skip]
+
+    return obs_data
+
+
+def get_normalized_bit(value, bit_index):
+    return (value >> bit_index) & 1
+
+
+def assign_values(data):
+    if data.dtype == float:
+        data[np.abs(data) >= np.abs(float_missing_value)] = float_missing_value
+        return np.array(data, dtype=ioda_float_type)
+    elif data.dtype == int:
+        data[np.abs(data) >= np.abs(int_missing_value)] = int_missing_value
+        return np.array(data, dtype=ioda_int_type)
+
+
+def get_WMO_satellite_ID(attrs_shortname):
+
+    if 'WSFM' in attrs_shortname:
+        WMO_sat_ID = WSF_WMO_sat_ID
+    else:
+        WMO_sat_ID = -1
+        print("could not determine satellite from filename: %s" % attrs_shortname)
+        sys.exit()
+
+    return WMO_sat_ID
+
+
+def init_obs_loc():
+    obs = {
+        ('brightnessTemperature', "ObsValue"): [],
+        ('brightnessTemperature', "ObsError"): [],
+        ('brightnessTemperature', "PreQC"): [],
+        ('satelliteIdentifier', metaDataName): [],
+        ('sensorChannelNumber', metaDataName): [],
+        ('latitude', metaDataName): [],
+        ('longitude', metaDataName): [],
+        ('dateTime', metaDataName): [],
+        ('sensorScanPosition', metaDataName): [],
+        ('solarZenithAngle', metaDataName): [],
+        ('solarAzimuthAngle', metaDataName): [],
+        ('sensorZenithAngle', metaDataName): [],
+        ('sensorAzimuthAngle', metaDataName): [],
+        ('sensorViewAngle', metaDataName): [],
+    }
+#       ('satelliteAscendingFlag', metaDataName): [],
```

Comment:
and need to also ask Steve if there is an ascending or descending flag in the data